### PR TITLE
[#33425] Fix leading item

### PR DESCRIPTION
**Pull request by b2z**
http://joomlacode.org/gf/project/joomla/tracker/?action=TrackerItemEdit&tracker_item_id=33425
### Before fix

float-bug
### After fix

after

### CODE REVIEW

**Comment by roland-d** [comment]
Let's say we remove the clearfix on the featured view as well. What is the final fix for getting the header to move left?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/3240">issues.joomla.org/joomla-cms/3240</a>.</sub>

**Comment by b2z** [comment]
> What is the final fix for getting the header to move left?

@roland-d  what do you mean?

**Comment by nternetinspired** [comment]
>  What is the final fix for getting the header to move left?

Don't float the preceding div.

**Comment by roland-d** [comment]
@b2z I meant how to come to a solution for this and get it tested and merged. The answer by @nternetinspired is Greek to me but perhaps it is enough for you to continue on this PR?

**Comment by b2z** [comment]
@nternetinspired can you please elaborate? There are no floating divs in the layout.

**Comment by nternetinspired** [comment]
This was the issue you originally posted @b2z . The float in your content caused the subsequent layout change, but that isn't a bug, that is how floats are supposed to work and is behaving as expected.

**Comment by sovainfo** [comment]
The original issue reported was a BC break by @nternetinspired removing obsolete html code. Turns out it wasn't obsolete and needed a clearfix tag (as provided in Featured blog, but left out for Category blog). Instead of admitting his mistake, he now claims doing the right thing in Featured blog was his mistake! 

With that he admits he tried to deceive intentionally by claiming to remove redundant code, but actually wants to change it. Obviously, that change is not desired. 

Consider it the responsibility of the Category/Featured blog to provide the expected output. Initially that was done using clearfix. Removing that makes the site look like having layout issues. That means site owners need to change all content because Joomla needed to break their site AGAIN!

See no need to continue that bad reputation!

**Comment by nternetinspired** [comment]
> With that he admits he tried to deceive intentionally by claiming to remove redundant code, but actually wants to change it. Obviously, that change is not desired.

I want to see bad, unnecessary and ill-considered code removed, yes. This is how improvements are made.

I've have never tried to deceive anyone here. Your accusation that I have intentionally tried to deceive is both untrue and unwarranted, as well as quite unbelievable. Let's stick to discussions about code and commits, rather than make personal attacks, shall we?

**Comment by brianteeman** [comment]
Please lets try not to make personal comments and comment ONLY on the code

On 15 May 2015 at 11:08, Seth Warburton <notifications@github.com> wrote:

> With that he admits he tried to deceive intentionally by claiming to
> remove redundant code, but actually wants to change it. Obviously, that
> change is not desired.
>
> I want to see bad, unnecessary and ill-considered code removed, yes. This
> is how improvements are made.
>
> I've have never tried to deceive anyone here. Your accusation that I have
> intentionally tried to deceive is both untrue and unwarranted, as well as
> quite unbelievable. Let's stick to discussions about code and commits,
> rather than make personal attacks, shall we?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/3240#issuecomment-102356518>.
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by sovainfo** [comment]
Just stating facts anybody can verify. As explained, the initial PR claimed to remove redundant code only. It turns out it is not redundant, applying the PR changed behaviour. Removing the so called redundant code had an undesired effect, which unfortunately wasn't detected. Can only hope it wouldn't have been committed if the inconsistency and changed behaviour was detected before. Assuming it would be corrected without changing its behaviour. 

With the easy fix as proposed by @b2z there is also no need to revert that PR. Just appy the tag where it is needed. 

Obviously, the original author considered it the responsibility of the blog to make sure articles are presented as desired and contents on articles don't mess up the blog layout.

Feel free to disagree on that, but don't change the rules of the game while playing!

Consider it deceiving claiming to remove redundant code while intending to change behaviour.  The behaviour shouldn't change.

**Comment by b2z** [comment]
So lets leave this clearfix :wink: Just synced the branch.

File: components/com_content/views/category/tmpl/blog.php [hunk 1]
```diff
@@ -54,7 +54,7 @@
 	<?php if (!empty($this->lead_items)) : ?>
 		<div class="items-leading clearfix">
 			<?php foreach ($this->lead_items as &$item) : ?>
-				<div class="leading-<?php echo $leadingcount; ?><?php echo $item->state == 0 ? ' system-unpublished' : null; ?>"
+				<div class="leading-<?php echo $leadingcount; ?><?php echo $item->state == 0 ? ' system-unpublished' : null; ? clearfix>"
```
Comment (bertmert) [inline components/com_content/views/category/tmpl/blog.php]: There's a "typo" in this line. Should be
```
<div class="leading-<?php echo $leadingcount; ?><?php echo $item->state == 0 ? ' system-unpublished' : null; ?> clearfix">
```

Comment (b2z) [inline components/com_content/views/category/tmpl/blog.php]: Thanks.Silly mistake :( Fixed.

**Comment by hitchblade** [comment]
Try to reproduce it, but I don't see the Problem.
(The Link is broken incidentally...) Sorry...<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/3240">issues.joomla.org/joomla-cms/3240</a>.</sub>

**Comment by Teck95** [comment]
Unfortunatelly I couldn't reproduce the issue. Given link is broken<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/3240">issues.joomla.org/joomla-cms/3240</a>.</sub>

**Comment by b2z** [comment]
@hitchblade @Teck95 what link do you mean, http://joomlacode.org? There is nothing there. The issue created here is the same as there. So basically nothing to test here ;)

**Comment by roland-d** [comment]
@b2z Not sure what link they mean either but they said that the issue cannot be reproduced.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/3240">issues.joomla.org/joomla-cms/3240</a>.</sub>

**Comment by Kubik-Rubik** [comment]
Thank you for your contribution but it has been decided that this is not something that will be included in the core of Joomla! at the moment.

**Comment by nternetinspired** [comment]
Who are you talking to @Kubik-Rubik? There are several contributors here…

Also, maybe good if you elaborate on the decision. Bug fixes will not be included in core at the moment??

**Pull request closed without merging**